client/client.go

```diff
@@ -251,16 +251,12 @@ func setParamsWithAuthorization(p *AuxiliaryParams, w *WrapperV2, params paramSe
 }
 
 // CreateAuthToken creates an auth token using the latest client.
-func (w *WrapperV2) CreateAuthToken(email, password string, p *AuxiliaryParams) (*auth_tokens.CreateAuthTokenOK, error) {
-	if w == nil {
-		return nil, microerror.Mask(clientV2NotInitializedError)
-	}
-
+func (w WrapperV2) CreateAuthToken(email, password string, p *AuxiliaryParams) (*auth_tokens.CreateAuthTokenOK, error) {
```

Comment:
I think pointer receiver is fine. Especially when you will need to e.g. use a mutex.
I just don't understand why this could ever be nil and would rather look for the root cause of the problem. Then just remove these nil checks.

Reply:
The root cause was that the client was not initialized correctly, when writing a unit test. If we think this is not dangerous, as it won't happen at normal runtime, then I'm fine with leaving out the nil checks.

Reply:
Now changed back to pointer receiver.

Reply:
Yes I think unit test should be just fixed. No need to handle that.